core/src/main/java/org/elasticsearch/search/aggregations/bucket/range/date/DateRangeAggregationBuilder.java

```diff
@@ -285,7 +281,25 @@ protected DateRangeAggregatorFactory innerBuild(SearchContext context, ValuesSou
             AggregatorFactory<?> parent, Builder subFactoriesBuilder) throws IOException {
         // We need to call processRanges here so they are parsed and we know whether `now` has been used before we make
         // the decision of whether to cache the request
-        Range[] ranges = processRanges(context, config);
+        RangeAggregator.Range[] ranges = processRanges(range -> {
+            DocValueFormat parser = config.format();
+            assert parser != null;
+            double from = range.getFrom();
+            double to = range.getTo();
+            String fromAsString = range.getFromAsString();
+            String toAsString = range.getToAsString();
+            if (fromAsString != null) {
+                from = parser.parseDouble(fromAsString, false, context.getQueryShardContext()::nowInMillis);
+            } else if (Double.isFinite(from)) {
+                from = parser.parseDouble(Long.toString((long) from), false, context.getQueryShardContext()::nowInMillis);
```

Comment:
That's true, but to me that's also what I would expect. If the `format` in the mapping is `strict_hour_minute_second` I would expect this to be the format of dates at index and at query time (if not specified otherwise). Note that the user could either use `strict_hour_minute_second||epoch_millis` in the mappings or use `"format" : "epoch_millis"` in the aggregation to still be able to supply the to and from as long.